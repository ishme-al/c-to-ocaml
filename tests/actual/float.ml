[@@@warning "-26"]
[@@@warning "-27"]
[@@@warning "-32"]
[@@@warning "-69"]

open Core

let rec set_at_index (lst : 'a list) (index : int) (value : 'a) : 'a list =
  match lst with
  | [] -> failwith "Index out of bounds"
  | hd :: tl ->
      if index = 0 then value :: tl else hd :: set_at_index tl (index - 1) value

let () =
  let a : float = 1. in
  let b : float = 2.3 in
  let c : float = Float.( + ) a b in
  let d : float = Float.( * ) a b in
  let e =
    if Float.( = ) b 0. then
      let e = 0. in
      e
    else
      let e = Float.( / ) a b in
      e
  in
  printf "c = %f\n" c;
  printf "d = %f\n" d;
  exit 0
