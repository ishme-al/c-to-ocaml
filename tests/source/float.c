int main() {
  float a = 1.;
  float b = 2.3;
  float c = a + b;
  float d = a * b;
  float e;
  if (b == 0.0) {
    e = 0.0;
  } else {
    e = a / b;
  }
  printf ("c = %f\n", c);
  printf ("d = %f\n", d);
  return 0;
}