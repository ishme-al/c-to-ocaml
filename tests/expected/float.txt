c = 3.300000
d = 2.300000
